src/model/av-media-description.ts

```diff
@@ -201,4 +216,48 @@ export class AvMediaDescription extends MediaDescription {
     });
     this.pts = this.pts.filter((existingPt) => allPtsToRemove.indexOf(existingPt) === -1);
   }
+
+  /**
+   * Add the given extension to this mline.
+   *
+   * @param extInfo - Information related to the extension to be added.
+   * @param extInfo.uri - The URI of the extension.
+   * @param extInfo.direction - An optional direction for the extension.
+   * @param extInfo.attributes - Option attributes for the extension.
+   * @param extInfo.id - An optional ID to use for the extension.  If no ID is passed, the first
+   * free (not already present in the map) ID will be used, starting at 1.  If the ID is
+   * invalid or already in use, an error is thrown.
+   */
+  addExtension({
+    uri,
+    direction,
+    attributes,
+    id,
+  }: {
+    uri: string;
+    direction?: ExtMapDirection;
+    attributes?: string;
+    id?: number;
+  }) {
+    // eslint-disable-next-line jsdoc/require-jsdoc
+    const getFirstFreeId = (): number => {
+      const inUseIds = [...this.extMaps.keys()].sort();
```

Comment:
It seems that we can avoid this extra sorting:

> const extMapSize = this.extMaps.size;
>     for (let idx = 1; idx < extMapSize + 1; idx += 1) {
>       if (!this.extMaps.has(idx)) {
>         return idx;
>       }
>     }
>     return extMapSize + 1;

Reply:
Good idea! Will tweak it.